amberdata/src/endpoint/token.ts

```diff
@@ -0,0 +1,48 @@
+import { Requester, Validator } from '@chainlink/external-adapter'
+import { ExecuteWithConfig, Config } from '@chainlink/types'
+
+export const Name = 'token'
+
+const customError = (data: any) => {
+  return Object.keys(data.payload).length === 0
+}
+
+export enum TokenEndpoints {
+  MarketCap = 'marketcap',
+  TotalSupply = 'totalsupply',
+}
+
+const paths: Record<TokenEndpoints, string> = {
+  [TokenEndpoints.MarketCap]: 'marketCapUSD',
+  [TokenEndpoints.TotalSupply]: 'totalSupply',
+}
+
+const customParams = {
+  base: ['base', 'from', 'coin'],
+  endpoint: true,
+}
+
+export const execute: ExecuteWithConfig<Config> = async (input, config) => {
+  const validator = new Validator(input, customParams)
+  if (validator.error) throw validator.error
+  const jobRunID = validator.validated.id
+
+  const coin = validator.validated.data.base
+  const endpoint: TokenEndpoints = validator.validated.data.endpoint
+  const path = paths[endpoint] || 'marketCapUSD'
```

Comment:
I don't think we should rely on `endpoint` like this at this level. See Nomics comments.